ceph/rados/core_workflows.py

```diff
@@ -2740,3 +2740,91 @@ def get_host_object(self, hostname):
 
         log.error(f"Could not find host object for host {hostname}")
         return None
+
+    def get_object_details_head(self, osd_id, object_name):
+        """
+        Method returns the object head in json format
+        Args:
+            osd_id: osd id number
+            object_name : object name
+        Return : Returns the object head details if exists or None
+
+        """
+        cmd_get_object = (
+            f"cephadm shell --name osd.{osd_id} -- ceph-objectstore-tool --data-path "
+            f"/var/lib/ceph/osd/ceph-{osd_id} --head --op list {object_name}"
+        )
+        acting_osd_node = self.fetch_host_node(daemon_type="osd", daemon_id=osd_id)
+        out_put = acting_osd_node.exec_command(sudo=True, cmd=cmd_get_object)
+        object_head = out_put[0].strip()
+        if object_name not in object_head:
+            log.error(
+                f"Not able to retrive the object head for the object-{object_name} "
+            )
+            return None
+        log.info(f"The object head of the {object_name} is {object_head}")
+        return object_head
+
+    def set_snapset_corrupt(self, osd_id, object_head):
```

Comment:
Is the method name correct? The method is actually clearing the snapset however the method name suggests setting the snapset corrupt.

Reply:
Modified the method description.